x-pack/plugins/spaces/server/plugin.ts

```diff
@@ -148,18 +148,28 @@ export class SpacesPlugin
       logger: this.log,
     });
 
-    const externalRouter = core.http.createRouter<SpacesRequestHandlerContext>();
-    initExternalSpacesApi({
-      externalRouter,
+    const buildFlavor = this.initializerContext.env.packageInfo.buildFlavor;
+    const router = core.http.createRouter<SpacesRequestHandlerContext>();
+    if (buildFlavor !== 'serverless') {
```

Comment:
I opted to make decisions as high-level as possible so we don't have to dig into each route to understand the impact of the build flavor.

Reply:
++, over time, if we still use `buildFlavor`, we might even re-organize our routes to make it easier to toggle them (e.g. `common` + `{non-serverless-name-tbd}-routes`)